client/utils/atomic.h

```diff
@@ -0,0 +1,74 @@
+/* atomic, T14.618-T14.618 $DVS:time$ */
+
+#ifndef _UTILS_ATOMIC_H
+#define _UTILS_ATOMIC_H
+
+#if defined(_WIN32) || defined(_WIN64)
+
+#include <stdatomic.h>
+
+/*******************  WINDOWS  *****************************
+Standard C11 atomic functions can take any atomic type without regards of the size
+so, for example, atomic_exchange can take atomic char or atomic uint64_t
+but on windows _Interlocked functions are divided by size:
+- _InterlockedExchange8 - _InterlockedExchange16 - _InterlockedExchange - _InterlockedExchange64 - _InterlockedExchange128.
+
+So you can easily map the atomic function in the correct _Interlocked function, for example:
+
+atomic_exchange_uint_least8(a,b) will be directly mapped to atomic_exchange(a,b) for C11 systems and
+it will be directly implemented with _Interlocked on Windows in stdatomic.h or here:
+
+example:
+
+uint_least8_t atomic_exchange_uint_least8(atomic_uint_least8_t *object, uint_least8_t desired)
+{
+	switch(sizeof(uint_least8_t)){
+		case 1:
+			return _InterlockedExchange8((char volatile*)object, desired);
+		case 2:
+			return _InterlockedExchange16((short volatile*)object, desired);
+		...continue...
+}
+**********************************************************/
+
+#define atomic_init_uintptr(a,b) atomic_init(a,b)
```

Comment:
it seems those definitions are the same

Reply:
changed it now =)

Reply:
@jonano614 merged with your implementation of stdatomic

Reply:
Hi, great! I will test it and merge.